packages/react-table/src/components/Table/base/types.tsx

```diff
@@ -22,6 +42,80 @@ export interface CellType {
   props?: object;
 }
 
+export interface TdSelectType {
+  /** The selectable variant */
+  variant?: 'checkbox' | 'radio';
+  /** Callback on select */
+  onSelect?: OnSelect;
+  /** Whether the cell is selected */
+  isSelected: boolean;
+  /** Whether to disable the selection */
+  disable?: boolean;
+  /** The row index */
+  rowIndex: number;
+  /** Additional props forwarded to select rowData */
+  props?: any;
+}
+
+export interface TdActionsType {
+  /** Cell actions */
+  items: IActions;
+  /** Whether to disable the actions */
+  disable?: boolean;
+  /** Actions dropdown position */
+  dropdownPosition?: DropdownPosition;
+  /** Actions dropdown direction */
+  dropdownDirection?: DropdownDirection;
+  /** Custom toggle for the actions menu */
+  actionsToggle?: (props: CustomActionsToggleProps) => React.ReactNode;
+}
+
+export interface TdExpandType {
+  isExpanded: boolean;
+  /** The row index */
+  rowIndex: number;
+  /** The column index */
+  columnIndex?: number;
+  /** On toggling the expansion */
+  onToggle?: OnCollapse;
+}
+
+export interface TdCompoundExpandType {
```

Comment:
All these interfaces that the consumer needs should be included in the MD file so the are documented.